tests/js/destructuring/destructuring.js

```diff
@@ -46,8 +46,31 @@ class A {
 
 const { a: { innerMember }, b: { anotherInnerMember } } = something;
 
+// these arguments should be destructured on multiple lines:
+function f2({ first: { inner1, inner2 }, second: { inner3, inner4 } }) {}
+
 // small enough for one line:
 const reducer1 = ({ first, second : { third } }) => combine(second, third);
 
 // these arguments should be destructured on multiple lines:
 const r2 = ({ a: { data1 }, b: { data2, data3 } }) => f(data1, data2, data3);
+
+const obj2 = {
+  // these arguments should be destructured on multiple lines:
+  func({ a: { info1, info2 }, b: { info3, info4 } }) {}
+};
+
+class A2 {
+  // these arguments should be destructured on multiple lines:
+  func({ a: { info1, info2 }, b: { info3, info4 } }) {}
+}
+
+// these catch arguments should be destructured on multiple lines:
+try {
+  // code
+} catch ({ first: { info1, info2 }, second: { info3, info4 } }) {
+  // code
+}
+
+// these destructured objects should be split into multiple lines:
+const { a: { innerData1, innerData2 } = {}, b: { innerData3 } = {} } = thing;
```

Comment:
For better readability of snapshot diffs, please create new test files instead of appending to existing ones.

Reply:
Done. Please feel free to propose changes if you think I used the wrong file name or should otherwise rearrange the new tests.